src/parser.rs

```diff
@@ -62,6 +78,18 @@ pub struct BasicParseError<'i> {
     pub location: SourceLocation,
 }
 
+impl<'i> fmt::Display for BasicParseError<'i> {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str(self.kind.description().as_str())
```

Comment:
That way this doesn't need the extra allocation.

Reply:
Also, it may be nice to expose self.location, but that may be fine as a follow-up.

Reply:
Done in c962d4a